llvm/lib/Transforms/Yk/StackMaps.cpp

```diff
@@ -99,6 +99,12 @@ class YkStackmaps : public ModulePass {
         // the offset of the stackmap entry will record the instruction after
         // the call, which is where we want to continue after deoptimisation.
         Bldr.SetInsertPoint(I->getNextNode());
+        CallInst &CI = cast<CallInst>(*I);
+        if (!CI.isIndirectCall() && CI.getCalledFunction()->getName() == YK_NEW_CONTROL_POINT) {
```

Comment:
The answer to this is probably "no", but is there any way we can assert that the last argument is what we expect here? Maybe we can check it's a pointer type or ...?

Reply:
I think we should be able to check it's a call to `frameaddr`.

Reply:
That would be great -- thanks!

Reply:
[f50ed9e](https://github.com/ykjit/ykllvm/pull/180/commits/f50ed9e27ea6378d406c57290b3de7e41d162a28)